mojo/services/view_manager/public/cpp/lib/view_manager_client_impl.cc

```diff
@@ -380,6 +380,9 @@ void ViewManagerClientImpl::OnViewDeleted(Id view_id) {
   View* view = GetViewById(view_id);
   if (view)
     ViewPrivate(view).LocalDestroy();
+  if (views_.size() == 0) {
+    delete this;
```

Comment:
this doesn't look right.  ViewManagerClientImpl has a couple different lifecycles - see the |delete_on_error| constructor parameter